src/Microsoft.IdentityModel.Tokens.Saml/Saml/SamlSecurityTokenHandler.ValidateToken.Internal.cs

```diff
@@ -90,14 +90,31 @@ internal async Task<ValidationResult<ValidatedToken>> ValidateTokenAsync(
 
             if (samlToken.Assertion.Conditions is not null)
             {
-                ValidationResult<DateTime?> tokenReplayValidationResult = Validators.ValidateTokenReplay(
-                    samlToken.Assertion.Conditions.NotOnOrAfter,
-                    samlToken.Assertion.CanonicalString,
-                    validationParameters,
-                    callContext);
+                ValidationResult<DateTime?> tokenReplayValidationResult;
+
+                try
+                {
+                    tokenReplayValidationResult = validationParameters.TokenReplayValidator(
```

Comment:
This calls `validationParameters.TokenReplayValidator` and before `Validators.ValidateTokenReplay` was called, intentional?